### DISCUSSION TITLE
Reusing `extraSelectors` in the `createFeature`

### DISCUSSION THREAD

**Discussion by alex-aghamyan-volo** [ngrx/platform#3774] [Q&A] [answered]
Hi there.
I was wondering is it possible to use selectors defined within `extraSelectors` in each other?

**Comment by markostanimirovic** [comment, marked as answer]
Hi @alex-aghamyan-volo 👋  

You can define `extraSelectors` factory in the following way:

```ts
export const counterFeature = createFeature({
  name: 'counter',
  reducer: createReducer({ count: 0 }),
  // 👇
  extraSelectors({ selectCount }) {
    const selectCountPlus1 = createSelector(
      selectCount,
      (count) => count + 1
    );
    const selectCountPlus2 = createSelector(
      // 👇
      selectCountPlus1,
      (count) => count + 1
    );

    return { selectCountPlus1, selectCountPlus2 };
  },
});
```

**Reply by alex-aghamyan-volo** [reply]
Thank you @markostanimirovic!

**Reply by a-boertien** [reply]
I ran into the same use case, and it wasn't obvious what the solution was -- thanks for this!

Maybe this should be added to the official docs:
https://ngrx.io/guide/store/feature-creators

**Reply by timdeschryver** [reply]
@a-boertien that's a good idea. Do you want to create PR for this?
I opened issue #4244 to track this.
